### DISCUSSION TITLE
Soft-wrap causes error

### DISCUSSION THREAD

**Discussion by glow12121** [helix-editor/helix#5814] [Q&A] [answered]
That's about all my config but I get an error listed below. I use Hx 22.12 on Linux.

```
[editor]
soft-wrap.enable = true
max-line-length = "70"

bufferline = "multiple"

[keys.normal]

[editor.cursor-shape]
```

```
$ hx
Bad config: unknown field `soft-wrap`, expected one of `scrolloff`, `scroll-lines`, `mouse`, `shell`, `line-number`, `cursorline`, `cursorcolumn`, `gutters`, `middle-click-paste`, `auto-pairs`, `auto-completion`, `auto-format`, `auto-save`, `idle-timeout`, `completion-trigger-len`, `auto-info`, `file-picker`, `statusline`, `cursor-shape`, `true-color`, `search`, `lsp`, `terminal`, `rulers`, `whitespace`, `bufferline`, `indent-guides`, `color-modes` for key `editor` at line 10 column 1
Press <ENTER> to continue with default config
```

In case I comment-out `soft-wrap.enable = true`, there's this error:

```
Bad config: newline in string found at line 3 column 26
``` 
(3, 26 is the end of the soft-wrap line.)

**Comment by pascalkuthe** [comment, marked as answer]
Did you compile from master or are you using the 22.12 release?
Softwrap only got merged a few days ago so it's only available if you build from master.

The `max-line-length` is only a language setting only so that would need to go into `languages.toml` also numbers should not be quoted so it should be `max-line-length = 70` instead

**Reply by glow12121** [reply]
I installed the editor from Fedora 37's repository.

Thank you - I understood what was the problem. Now can't wait to start using soft-wrapping for markdown.

By the way, can `max-line-length` be applied to every language?

**Reply by pascalkuthe** [reply]
> By the way, can max-line-length be applied to every language?

Not yet, we want to add a setting for that but the softwrap PR was already a huge effort and I kept any optional feature out to keep the scope in check.

**Reply by glow12121** [reply]
I see!
